redis.conf

```diff
@@ -1578,6 +1578,18 @@ lua-time-limit 5000
 #
 # cluster-allow-reads-when-down no
 
+# Clusters can announce a hostname to be used to connect to in addition
+# to an IP address. This is a common use case for applications that need
+# to use TLS Server Name Indication (SNI) or dealing with DNS based
+# routing. Where this hostname is displayed is controlled by a second config,
+# cluster-prefer-hostname. When set to yes, the hostname will show up in
+# MOVED/ASKING requests as well as the first field in CLUSTER SLOTS. When set
+# to no, the default, the hostname will only show up in the fourth field of
+# CLUSTER SLOTS as additional metadata.
+#
+# cluster-announce-hostname ""
+# cluster-prefer-hostname no
```

Comment:
Do we really need `cluster-prefer-hostname`?

An alternative is to treat `cluster-announce-hostname` like the other `cluster-announce-` parameters and simply use the hostname instead of the IP whenever it's available.

This is not mutually exclusive to the idea of extending `CLUSTER SLOTS` with additional meta-data.

Reply:
I'm not sure it's purely analogous to to the announce IPs, this is declaring an alternative way to find the node. My original thought for this argument is that in the TLS case, you may still want to connect to the node's IP, and if the hostname is available you can use that for hostname validation. You may not want to take a dependency on DNS or some other host discovery service.

Reply:
This is a valid point, but in practice I think these two usually go hand in hand. That is, if you're providing an IP instead of a hostname, you'd probably turn off certificate validation rather than say "hey, here's the real hostname I expect but did not use for the connect".

I'm OK either way.

Reply:
I'm okay either way as well. The original thought here was that AWS we've seen some annoyances with DNS caching, so some users only went to have the hostname for SNI but not for anything else. This flag mostly covers that case.